pkg/strategy/grid2/recover.go

```diff
@@ -319,3 +310,173 @@ func addOrdersIntoPinOrderMap(pinOrders PinOrderMap, orders []types.Order) error
 
 	return nil
 }
+
+func (s *Strategy) verifyFilledTwinGrid(pins []Pin, twinOrders TwinOrderMap, filledOrders []types.Order) error {
+	s.debugLog("verifying filled grid - pins: %+v", pins)
+	s.debugLog("verifying filled grid - open twin orders:\n%s", twinOrders.String())
+	s.debugOrders("verifying filled grid - filled orders", filledOrders)
+
+	if err := s.addOrdersIntoTwinOrderMap(twinOrders, filledOrders); err != nil {
+		return errors.Wrapf(err, "verifying filled grid error when add orders into twin order map")
+	}
+
+	s.debugLog("verifying filled grid - filled twin orders:\n%+v", twinOrders.String())
+
+	for i, pin := range pins {
+		if i == 0 {
```

Comment:
Could you leave a comment here describing it's because sell price is used as the key of twin order map?